src/probnum/diffeq/odefiltsmooth/kalman_odesolution.py

```diff
@@ -124,3 +124,9 @@ def sample(
 
         samples = self.kalman_posterior.sample(locations=t, size=size)
         return np.array([self.proj_to_y @ sample for sample in samples])
+
+    @property
+    def filter_solution(self):
```

Comment:
Where is this used / what is the use-case for this property?

Reply:
extracting the filtering ODE solution out of the kalman posterior. `SmoothingPosterior` has an attribute `filter_posterior`, which is accessed directly. Since `KalmanODESolution` wraps around `KalmanPosterior` objects, there is no un-awkward way of accessing the filter solution (the only way would be to manually project the states out of self.kalman_poisterior.filter_posterior) which is avoided with this simple property.


But seeing this, I will add a useful error if `self.kalmanposterior` has no `filterposterior`, i.e. if it is the FilteringPosterior.

Reply:
respectively return `self`: 

```python
    @property
    def filter_solution(self):

        if isinstance(self.kalman_posterior, pnfs.FilteringPosterior):
            return self

        # else: self.kalman_posterior is a SmoothingPosterior object, which has the field filter_posterior.
        return KalmanODESolution(
            kalman_posterior=self.kalman_posterior.filter_posterior
        )
```

Reply:
Sounds good, thanks for the explanation :)